src/main/java/org/spongepowered/api/text/title/TitleBuilder.java

```diff
@@ -24,91 +24,311 @@
  */
 package org.spongepowered.api.text.title;
 
-import org.spongepowered.api.text.message.Message;
+import com.google.common.base.MoreObjects;
+import com.google.common.base.Objects;
+import com.google.common.base.Optional;
+import org.spongepowered.api.text.Text;
+
+import javax.annotation.Nullable;
 
 /**
- * Represents a builder interface to create immutable {@link Title}
- * configurations.
+ * Represents a builder class to create immutable {@link Title} configurations.
+ *
+ * @see Title
  */
-public interface TitleBuilder {
+public class TitleBuilder {
+
+    @Nullable protected Text title;
+    @Nullable protected Text subtitle;
+    @Nullable protected Integer fadeIn;
+    @Nullable protected Integer stay;
+    @Nullable protected Integer fadeOut;
+    protected boolean clear;
+    protected boolean reset;
+
+    /**
+     * Constructs a new empty {@link TitleBuilder}.
+     */
+    public TitleBuilder() {
+    }
+
+    /**
+     * Constructs a new {@link TitleBuilder} with the properties of the given
+     * {@link Title} as initial values.
+     *
+     * @param title The title to copy the values from
+     */
+    protected TitleBuilder(Title title) {
```

Comment:
This message was created automatically by mail delivery software.

A message that you sent could not be delivered to one or more of its
recipients. This is a temporary error. The following address(es) deferred:

  adam@eq2.serveftp.net
    Domain eq2.co.uk has exceeded the max emails per hour (66/60 (110%)) allowed.  Message will be reattempted later

------- This is a copy of the message, including all the headers. ------
Received: from o5.sgmail.github.com ([192.254.113.10]:5748)
	by web3.cov.yesit.co.uk with esmtps (TLSv1.2:DHE-RSA-AES128-GCM-SHA256:128)
	(Exim 4.84)
	(envelope-from <bounces+848413-d0b0-adam=eq2.co.uk@sgmail.github.com>)
	id 1YNea0-0003YD-0L
	for adam@eq2.co.uk; Tue, 17 Feb 2015 09:36:32 +0000
DKIM-Signature: v=1; a=rsa-sha1; c=relaxed; d=github.com; 
	h=from:reply-to:to:cc:in-reply-to:references:subject:mime-version:content-type:content-transfer-encoding:list-id:list-archive:list-post:list-unsubscribe; 
	s=s20150108; bh=jZHAD1tv6EMEiHuBbD5o7+zZH6U=; b=PTGQ0HX/89GHVldd
	tAtN3R0s403hfU0qjBovIFA1XEnBceSHSP9uJbl2nzvU7rCflEVPOnU78jx+oWmf
	1lUdy+ozPSiSguuNNFIba0g8V3PJ0LJSeUGbn/6jrvfyZ7/PJJ9r6E+Ff79Ykj6z
	giDJC6TxiXZI9c7wz0TydpAN1s0=
Received: by filter0213p1mdw1.sendgrid.net with SMTP id filter0213p1mdw1.9495.54E30B992
        2015-02-17 09:36:30.350251544 +0000 UTC
Received: from github-smtp2a-ext-cp1-prd.iad.github.net (github-smtp2a-ext-cp1-prd.iad.github.net [192.30.253.16])
	by ismtpd-025 (SG) with ESMTP id 14b96e5622f.2f85.55f9b9
	for <adam@eq2.co.uk>; Tue, 17 Feb 2015 09:36:30 +0000 (UTC)
Date: Tue, 17 Feb 2015 01:36:30 -0800
From: gratimax <notifications@github.com>
Reply-To: SpongePowered/SpongeAPI <reply@reply.github.com>
To: SpongePowered/SpongeAPI <SpongeAPI@noreply.github.com>
Cc: Mumfrey <adam@eq2.co.uk>
Message-ID: <SpongePowered/SpongeAPI/pull/425/r24803091@github.com>
In-Reply-To: <SpongePowered/SpongeAPI/pull/425@github.com>
References: <SpongePowered/SpongeAPI/pull/425@github.com>
Subject: Re: [SpongeAPI] [WIP] Text API 2 (#425)
Mime-Version: 1.0
Content-Type: multipart/alternative;
 boundary="--==_mimepart_54e30b9e1e22e_de53feb6fb732bc100854f";
 charset=UTF-8
Content-Transfer-Encoding: 7bit
Precedence: list
X-GitHub-Sender: gratimax
X-GitHub-Recipient: Mumfrey
X-GitHub-Reason: comment
List-ID: SpongePowered/SpongeAPI <SpongeAPI.SpongePowered.github.com>
List-Archive: https://github.com/SpongePowered/SpongeAPI
List-Post: <mailto:reply@reply.github.com>
List-Unsubscribe: <mailto:unsub+001180bf2e9bca7ba0be65d91e14fc929423f4f93083229292cf0000000110facd9e92a170ce017a7713@reply.github.com>,
 <https://github.com/notifications/unsubscribe/ABGAv78XglFt1kjfysSub9apgaJDrn3Zks5nswMegaJpZM4DZr_C>
X-Auto-Response-Suppress: All
X-GitHub-Recipient-Address: adam@eq2.co.uk
X-SG-EID: VzkE0stiinYMPYWPWpnZ4OvH/Ru8MrIuYzb7JqhnFX9s7W8+CsMFL1dS81l3kiklch9I+0D/19ScSO
 06bztosEW+yEknCl2BFAI3wP3SAdzOArUHSEzss8A3NzUqFTuJYwMypRrjsj86teZ+t0Mdbkm2I6eA
 kb3VzBy6mnH7Xeg=

----==_mimepart_54e30b9e1e22e_de53feb6fb732bc100854f
Content-Type: text/plain;
 charset=UTF-8
Content-Transfer-Encoding: 7bit

> +    protected boolean clear;
> +    protected boolean reset;
> +
> +    /**
> +     * Constructs a new empty {@link TitleBuilder}.
> +     */
> +    public TitleBuilder() {
> +    }
> +
> +    /**
> +     * Constructs a new {@link TitleBuilder} with the properties of the given
> +     * {@link Title} as initial values.
> +     *
> +     * @param title The title to copy the values from
> +     */
> +    protected TitleBuilder(Title title) {

why protected

---
Reply to this email directly or view it on GitHub:
https://github.com/SpongePowered/SpongeAPI/pull/425/files#r24803091
----==_mimepart_54e30b9e1e22e_de53feb6fb732bc100854f
Content-Type: text/html;
 charset=UTF-8
Content-Transfer-Encoding: 7bit

<p>In <a href="https://github.com/SpongePowered/SpongeAPI/pull/425#discussion_r24803091">src/main/java/org/spongepowered/api/text/title/TitleBuilder.java</a>:</p>
<pre style='color:#555'>&gt; +    protected boolean clear;
&gt; +    protected boolean reset;
&gt; +
&gt; +    /**
&gt; +     * Constructs a new empty {@link TitleBuilder}.
&gt; +     */
&gt; +    public TitleBuilder() {
&gt; +    }
&gt; +
&gt; +    /**
&gt; +     * Constructs a new {@link TitleBuilder} with the properties of the given
&gt; +     * {@link Title} as initial values.
&gt; +     *
&gt; +     * @param title The title to copy the values from
&gt; +     */
&gt; +    protected TitleBuilder(Title title) {
</pre>
<p>why protected</p>

<p style="font-size:small;-webkit-text-size-adjust:none;color:#666;">&mdash;<br>Reply to this email directly or <a href="https://github.com/SpongePowered/SpongeAPI/pull/425/files#r24803091">view it on GitHub</a>.<img alt="" height="1" src="https://github.com/notifications/beacon/ABGAv1k2GOvwlRXKR3DxfSVsxm0RqGd1ks5nswMegaJpZM4DZr_C.gif" width="1" /></p>
<div itemscope itemtype="http://schema.org/EmailMessage">
  <div itemprop="action" itemscope itemtype="http://schema.org/ViewAction">
    <link itemprop="url" href="https://github.com/SpongePowered/SpongeAPI/pull/425/files#r24803091"></link>
    <meta itemprop="name" content="View Pull Request"></meta>
  </div>
  <meta itemprop="description" content="View this Pull Request on GitHub"></meta>
</div>

----==_mimepart_54e30b9e1e22e_de53feb6fb732bc100854f--